docs/source/montage_creation.md

```diff
@@ -0,0 +1,150 @@
+```
+
+This will create a montage from all TIFF files in folder `/data`. Each TIFF file is read as a multi-level image, and
+level 2 is read for creating the montage.
+
+The `--width` argument determines the width in pixel of the output image. The height of the output image is
+
+This will output two images, `montage1/montage.jpg` and `montage1/montage.png`.
+
+Here's an example how this could look like for a folder with 6 images, `0.tiff` through `5.tiff`:
+
+![image](images/montage_from_folder.png)
+
+## Creating Montages From a `dataset.csv` File
+
+If the montage creation script is only pointed to a folder, without providing a glob pattern,
+it assumes that a file `dataset.csv` is present. A montage will be created from only the images
+listed in `dataset.csv`. In addition, an optional `label` column will be added to the text that is
+overlayed onto the images itself.
+
+The dataset file should be a CSV file, with each row corresponding to a single image.
+When working with a `dataset.csv` file, the following columns are handled:
+
+| Column name | Contents | Required? |
+|---|---|---|
+| `image` | The path of the image that should be loaded | Required |
+| `slide_id` | A unique identifier for the slide | Required |
+| `label` | An additional string that will be placed on the montage, This could be `0`, `1`, `tumour`, ... | Optional |
+| `mask` | The path of an additional image that will rendered next to the image given in `image` | Optional |
+
+Consider this example dataset file:
+
+```text
+image,slide_id,label
+2.tiff,ID 2,Label 2
+3.tiff,ID 3,Label 3
+4.tiff,ID 4,Label 4
+5.tiff,ID 5,Label 5
+```
+
+Run montage creation with the following command:
+
+```shell
+python src/health_cpath/scripts/create_montage.py --dataset /data --level 2 --width 1000 --output_path montage1
+```
+
+This would produce (assuming that the images `2.tiff`, `3.tiff`, `4.tiff`, and `5.tiff` are present in the folder
+`/data`) a montage similar to this one:
+
+![image](images/montage_from_dataset.png)
+
+### Using Inclusion or Exclusion Lists
+
+When creating montages from a `dataset.csv` file, it is possible to create montages from only a specific subset
+of rows, or all rows apart from those in a given list.
+
+- Use the `--exclude_by_slide_id exclude.txt` argument to point to a file with a list of slide IDs that should be
+   excluded from the montage.
+- Use the `--include_by_slide_id include.txt` argument to point to a file with a list of slide IDs for which
+   the montage should be created.
+
+The files `exclude.txt` and `include.txt` should contain one slide ID per line.
+
+## Other Commandline Options
+
+- Use `--width=1000` to set the width of the output image. The height of the output image is determined
+  automatically. Mind that the default value is 60_000, suitable for several hundreds of images. If you want to try
```

Comment:
Not sure if 60_000 is width of a single image or whole montage?

Reply:
60_000 is the width of the whole montage, i've slightly rephrased, hope it's clearer now.